ddtrace/contrib/flask/patch.py

```diff
@@ -128,6 +129,13 @@ def _request_span_modifier(self, span, environ):
         req_body = None
         if config._appsec_enabled and request.method in _BODY_METHODS:
             content_type = request.content_type
+            wsgi_input = environ.get("wsgi.input", "")
+
+            # Copy wsgi input if not seekable
+            if not hasattr(wsgi_input, "seekable") or not wsgi_input.seekable():
+                body = wsgi_input.read()
+                environ["wsgi.input"] = BytesIO(body)
```

Comment:
is `BytesIO` correct for Python 2?

```
Again, all objects referred to in this specification as “strings” must be of type str or StringType, and must not be of type unicode or UnicodeType. And, even if a given platform allows for more than 8 bits per character in str/StringType objects, only the lower 8 bits may be used, for any value referred to in this specification as a “string”.

For values referred to in this specification as “bytestrings” (i.e., values read from wsgi.input, passed to write() or yielded by the application), the value must be of type bytes under Python 3, and str in earlier versions of Python.
```

specifically:

> the value must be of type bytes under Python 3, and str in earlier versions of Python.

Reply:
Seems like Werkzeug since 0.16 at least (Flask<1.0.0) does coerce the `wsgi.input` into a `BytesIO` so this should be safe

https://github.com/pallets/werkzeug/blob/2b2c4c3dd3cf7389e9f4aa06371b7332257c6289/src/werkzeug/wsgi.py#L202-L235